common/scala/src/main/scala/whisk/core/entity/ActivationId.scala

```diff
@@ -42,7 +42,7 @@ import whisk.http.Messages
  */
 protected[whisk] class ActivationId private (private val id: java.util.UUID) extends AnyVal {
   def asString = toString
-  override def toString = id.toString.replaceAll("-", "")
+  override def toString = id.toString.filterNot(_ == '-')
```

Comment:
For some reasons, we suspect that the `toString` and `asString` methods are run rather often. So it's probably no pre-mature optimisation to compute the filter upon construction and have a `val idString` that is returned instead of computing on the fly.

Reply:
You'll have to remove `AnyVal` and make the activation id not a value to do that.